File: internal/tui/components/detail.go
package components

import (
	"strings"

	"github.com/charmbracelet/lipgloss"
	"github.com/mailersend/mailersend-cli/internal/tui/theme"
)

var (
	detailTitleStyle = lipgloss.NewStyle().
				Bold(true).
				Foreground(theme.Primary).
				MarginBottom(1)

	detailLabelStyle = lipgloss.NewStyle().
				Foreground(theme.Muted).
				Width(20)

	detailValueStyle = lipgloss.NewStyle().
				Foreground(theme.Text)

	detailHintStyle = lipgloss.NewStyle().
			Foreground(theme.Muted).
			Italic(true).
			MarginTop(1)
)

// DetailRow represents a key-value pair for display.
type DetailRow struct {
	Label string
	Value string
}

// DetailPanel displays detailed information about an item.
type DetailPanel struct {
	title  string
	rows   []DetailRow
	width  int
	height int
}

// NewDetailPanel creates a new detail panel.
func NewDetailPanel() DetailPanel {
	return DetailPanel{}
}

// SetTitle sets the panel title.
func (d *DetailPanel) SetTitle(title string) {
	d.title = title
}

// SetRows sets the detail rows.
func (d *DetailPanel) SetRows(rows []DetailRow) {
	d.rows = rows
}

// SetSize sets the panel dimensions.
func (d *DetailPanel) SetSize(width, height int) {
	d.width = width
	d.height = height
}

// View renders the detail panel.
func (d DetailPanel) View() string {
	var b strings.Builder

	// Title
	b.WriteString(detailTitleStyle.Render(d.title))
	b.WriteString("\n\n")

	// Rows
	for _, row := range d.rows {
		if row.Label == "" && row.Value == "" {
			b.WriteString("\n")
			continue
		}
		if row.Label == "" {
			value := detailValueStyle.Render(row.Value)
			b.WriteString(detailLabelStyle.Render("") + " " + value + "\n")
			continue
		}
		label := detailLabelStyle.Render(row.Label + ":")
		value := detailValueStyle.Render(row.Value)
		b.WriteString(label + " " + value + "\n")
	}

	// Hint
	b.WriteString("\n")
	b.WriteString(detailHintStyle.Render("Press Esc or Backspace to go back"))

	return b.String()
}

File: src/tui/components/detail.rs
use ratatui::style::{Modifier, Style};
use ratatui::text::{Line, Span};

use super::pad_right;
use crate::tui::theme;

pub struct DetailRow {
    pub label: String,
    pub value: String,
}

impl DetailRow {
    pub fn new(label: impl Into<String>, value: impl Into<String>) -> Self {
        Self {
            label: label.into(),
            value: value.into(),
        }
    }
}

pub struct DetailPanel {
    title: String,
    rows: Vec<DetailRow>,
}

impl DetailPanel {
    pub fn new() -> Self {
        Self {
            title: String::new(),
            rows: Vec::new(),
        }
    }

    pub fn set_title(&mut self, title: impl Into<String>) {
        self.title = title.into();
    }

    pub fn set_rows(&mut self, rows: Vec<DetailRow>) {
        self.rows = rows;
    }

    pub fn lines(&self) -> Vec<Line<'static>> {
        let label_style = Style::new().fg(theme::MUTED);
        let value_style = Style::new().fg(theme::TEXT);

        let mut lines: Vec<Line<'static>> = vec![
            Line::styled(
                self.title.clone(),
                Style::new().fg(theme::PRIMARY).add_modifier(Modifier::BOLD),
            ),
            Line::raw(""),
        ];

        for row in &self.rows {
            if row.label.is_empty() && row.value.is_empty() {
                lines.push(Line::raw(""));
                continue;
            }
            let label = if row.label.is_empty() {
                pad_right("", 20)
            } else {
                pad_right(&format!("{}:", row.label), 20)
            };
            lines.push(Line::from(vec![
                Span::styled(label, label_style),
                Span::raw(" "),
                Span::styled(row.value.clone(), value_style),
            ]));
        }

        lines.push(Line::raw(""));
        lines.push(Line::styled(
            "Press Esc or Backspace to go back",
            Style::new().fg(theme::MUTED).add_modifier(Modifier::ITALIC),
        ));

        lines
    }
}
